File: node/controllers/update-unit-in-army-roster-controller.ts
import { Request, Response } from 'express';
import { updateUnitInArmyRosterInDb } from '../models/update-unit-in-army-roster-model';

export async function updateUnitInArmyRosterController(
    req: Request,
    res: Response
) {
    try {
        const rosterId = Number(req.body.roster_id) || 0;
        const isAdding = Boolean(req.body.is_adding);
        const unitData = await updateUnitInArmyRosterInDb(rosterId, isAdding);
        res.statusCode = 200;
        res.json(unitData);
    } catch (error) {
        res.statusCode = 500;
        res.send();
    }
}

File: python/controllers/update_unit_in_army_roster_controller.py
import json
from operator import pos
from flask import request
import json

from models.update_unit_in_army_roster_model import update_unit_in_roster_model


def update_unit_roster_controller():
    body = request.get_json()
    if body is None:
        return []
    roster_id = body.get("roster_id", 0)
    is_adding = body.get("is_adding", False)
    if roster_id is 0:
        return []
    return update_unit_in_roster_model(roster_id, is_adding)
